setup.py

```diff
@@ -19,19 +19,93 @@
 from setuptools import setup, find_packages
 import versioneer
 
+import os
+import warnings
 import codecs
 
+
+def abspath(file):
+    return os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                        file)
+
+
+def dynamic_author_list():
+    """Generate __authors__ from AUTHORS
+
+    This function generates authors.py that contains the list of the
+    authors from the AUTHORS file. This avoids having that list maintained in
+    several places. Note that AUTHORS is sorted chronologically while we want
+    __authors__ in authors.py to be sorted alphabetically.
+
+    The authors are written in AUTHORS as bullet points under the
+    "Chronological list of authors" title.
+    """
+    authors = []
+    with codecs.open(abspath('AUTHORS'), encoding='utf-8') as infile:
+        # An author is a bullet point under the title "Chronological list of
+        # authors". We first want move the cursor down to the title of
+        # interest.
+        for line_no, line in enumerate(infile, start=1):
+            if line.rstrip() == "Chronological list of authors":
+                break
+        else:
+            # If we did not break, it means we did not find the authors.
+            raise IOError('EOF before the list of authors')
+        # Skip the next line as it is the title underlining
+        line = next(infile)
+        line_no += 1
+        if line[:4] != '----':
+            raise IOError('Unexpected content on line {0}, '
+                          'should be a string of "-".'.format(line_no))
+        # Add each bullet point as an author
+        for line in infile:
+            if line.strip()[:2] == '- ':
+                # This is a bullet point, so it should be an author name.
+                name = line.strip()[2:].strip()
+                # Remove any @-handles in "A.U. Thor (@donnergott)"
+                name = name.split("(")[0].strip()
+                authors.append(name)
+
+    # So far, the list of authors is sorted chronologically. We want it
+    # sorted alphabetically of the last name.
+    authors.sort(key=lambda name: name.split()[-1])
+
+    # Write the authors.py file.
+    out_path = abspath('MDAnalysisData/authors.py')
+    with codecs.open(out_path, 'w', encoding='utf-8') as outfile:
+        # Write the header
+        header = '''\
+#-*- coding:utf-8 -*-
+
+# This file is generated from the AUTHORS file during the installation process.
+# Do not edit it as your changes will be overwritten.
+'''
+        print(header, file=outfile)
+
+        # Write the list of authors as a python list
+        template = u'__authors__ = [\n{}\n]'
+        author_string = u',\n'.join(u'    u"{}"'.format(name)
+                                    for name in authors)
+        print(template.format(author_string), file=outfile)
```

Comment:
This code was copied from MDAnalysis setup.py (published under GPL v2+). It _cannot_ be used in this package under BSD-3 unless the original authors @jbarnoud @tylerjereddy @lilyminium re-license the code.

Reply:
Is written permission via comment here ok? Fine by me (I prefer BSD-3 or even MIT anyway).

Reply:
I would take written permission here as acceptable.

If all three authors agree, I'd add a comment to the code stating explicitly that the code was re-licensed with a link to the PR/comments.

Reply:
happy with MIT/BSD-3 also!

Reply:
I am OK with the change.

Reply:
Thank you all, will update the PR.